### PR TITLE
Add basic atmos products for GEFS and refactor resources

### LINKED ISSUE #823: Add atm ensemble product generation
**Description**
As we move towards the merger of GEFS into the GFS system, global workflow needs to be updated to produce the same products the current GEFS system is creating.

**Requirements**
Some post scripts will need to be updated to accommodate ensemble products, while others may need to be added based on GEFS scripts. The end result will need to be able to produce both GFS and GEFS products, depending on the member number.

Products:
- [ ] Member lat-lon grib files
- [ ] Mean and spread grib files
- [ ] Gempak files

Production of ensstat files can be skipped for now.

**Acceptance Criteria (Definition of Done)**
Given master grib files from an ensemble forecast, be able to run a series of post-processing jobs to produce the above products as GEFS does.

**Dependencies**
Reorganize COM (https://github.com/NOAA-EMC/global-workflow/issues/761)
Update workflow setup scripts (https://github.com/NOAA-EMC/global-workflow/issues/789) - Closed
Add ensemble free-forecast (#822) - Done 
Move post scripts from UPP under global-workflow (#769) - Done
Create flexible product script to replace old post script (#270) [may be concurrent]

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
# Description
Adds basic atmosphere products for individual GEFS members (no mean or spread).

To facilitate the addition of GEFS parameter lists, the existing GFS lists are renamed. They are also relocated to `parm/products/` directory instead of `parm/post/`. GEFS v12 uses different parameter lists for different resolutions, but for this implementation, the 0p25 lists are used for all resolutions. However, to hedge against the final configuration, all of the GEFS parameter lists have been added to parm. Implementation of different parameter lists for each resolution will take additional refactoring of the atmos_products script, as it currently assumes all resolutions use the same lists.

The new GEFS rocoto task is implemented as two loops of metatasks (made possible by PR #2189). The member is set at the rocoto level, so the setting in `config.base` had to be updated to accept a pass-through. The generation of a forecast hour list is moved up to its own function in `rocoto.tasks`, separate from the post groups which are not used. This can be used as a model for future metatasks (or refactoring of old ones). `rocoto_viewer.py` does function with the nested metatasks, though only one level of folding is available and the metatask name is that of the lower metatask (though it encompasses the entire outer metatask).

Additionally, the GEFS `config.resources` had diverged from the GFS version, which included not having a section for the atm products job. This led to the refactoring of the GFS `config.resources`, including switching to using a `case` statement instead of sequential `elif`. This refactored script was then copied over to GEFS, replacing the old one, with all jobs not currently implemented for GEFS removed (except wave post, whose inclusion is imminent). New blocks can be copied over from GFS as they are added.

Resolves #823

# Type of change
- New feature (adds functionality)
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
- Cycled test on Orion
- GEFS test on Orion

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by WalterKolczynski-NOAA** [comment]
@WenMeng-NOAA Sample output is located in
```
/work2/noaa/global/wkolczyn/noscrub/global-workflow/atm_prod_gefs/gefs.20210323/12
```

These are still using the old GFS names, which will be updated in the future.

File: parm/config/gfs/config.resources [hunk 1]
```diff
@@ -32,620 +32,589 @@ step=$1
...
     export nth_wavepostsbs=1
-    npe_node_wavepostsbs=$(echo "${npe_node_max} / ${nth_wavepostsbs}" | bc)
-    export npe_node_wavepostsbs
+    export npe_node_wavepostsbs=$(( npe_node_max / nth_wavepostsbs ))
     export NTASKS=${npe_wavepostsbs}
     export memory_wavepostsbs="10GB"
-    export memory_wavepostsbs_gfs="10GB"
-
-elif [[ ${step} = "wavepostbndpnt" ]]; then
+    export memory_wavepostsbs_gfs="1GB"
```
Comment (aerorahul) [inline parm/config/gfs/config.resources]:
```suggestion
    export memory_wavepostsbs_gfs="10GB"
```

File: parm/config/gfs/config.resources [hunk 2]
```diff
@@ -791,313 +760,312 @@ elif [[ "${step}" = "metp" ]]; then
...
-
+  "init")
     export wtime_init="00:30:00"
     export npe_init=24
     export nth_init=1
     export npe_node_init=6
-    export memory_init="70G"
-
-elif [[ ${step} = "init_chem" ]]; then
+    export memory_init="70"
```
Comment (aerorahul) [inline parm/config/gfs/config.resources]:
```suggestion
    export memory_init="70GB"
```

**Review by aerorahul — CHANGES_REQUESTED** [review]
Looks good.
I did not do a careful review of the refactored `config.resources` but found a few isolated misses when specifying memory.  Most of the memory specs are consistent and clear `GB`, in some places a number is noticed e.g. `3072`.  Perhaps that should be made clear as `3GB` or whatever the unit needs to be.

File: parm/config/gfs/config.resources (hunk 1 above)
Comment (WalterKolczynski-NOAA) [reply parm/config/gfs/config.resources]: Fixed

File: parm/config/gfs/config.resources (hunk 2 above)
Comment (WalterKolczynski-NOAA) [reply parm/config/gfs/config.resources]: Fixed

**Comment by WalterKolczynski-NOAA** [comment]
> Looks good. I did not do a careful review of the refactored `config.resources` but found a few isolated misses when specifying memory. Most of the memory specs are consistent and clear `GB`, in some places a number is noticed e.g. `3072`. Perhaps that should be made clear as `3GB` or whatever the unit needs to be.

I searched for `3072` and didn't find any that didn't have an `M` afterward anymore, so I think these have all been handled.

**Comment by WalterKolczynski-NOAA** [comment]
Hera failed due to radmon_angle.x returning a non-zero value. I just confirmed the values in `config.resources` are identical, so it isn't something I changed.

```
+ radmon_verf_angle.sh[141]: prep_step
+ radmon_verf_angle.sh[143]: ctr=14
+ radmon_verf_angle.sh[145]: [[ anl == \a\n\l ]]
+ radmon_verf_angle.sh[146]: data_file=atms_npp_anl.2021122100.ieee_d
+ radmon_verf_angle.sh[147]: ctl_file=atms_npp_anl.ctl
+ radmon_verf_angle.sh[148]: angl_ctl=angle.atms_npp_anl.ctl
+ radmon_verf_angle.sh[155]: angl_file=
+ radmon_verf_angle.sh[156]: [[ 0 -eq 1 ]]
+ radmon_verf_angle.sh[161]: [[ -f input ]]
+ radmon_verf_angle.sh[161]: rm input
+ radmon_verf_angle.sh[163]: nchanl=-999
+ radmon_verf_angle.sh[164]: cat
+ radmon_verf_angle.sh[182]: startmsg

WARNING: $jlogfile variable not defined. Message intended for the jlogfile:
  01/12 19:31:03Z verfrad.57455-radmon_angle.x started

+ radmon_verf_angle.sh[183]: ./radmon_angle.x
+ radmon_verf_angle.sh[1]: postamble radmon_verf_angle.sh 1705087854 1
+ preamble.sh[68]: set +x
End radmon_verf_angle.sh at 19:31:03 with error code 1 (time elapsed: 00:00:09)
```

File: parm/config/gfs/config.resources [hunk 3]
```diff
@@ -791,313 +760,312 @@ elif [[ "${step}" = "metp" ]]; then
...
     export npe_gempak=2
     export npe_gempak_gfs=28
     export npe_node_gempak=2
     export npe_node_gempak_gfs=28
     export nth_gempak=1
     export memory_gempak="4GB"
-    export memory_gempak_gfs="2GB"
-
-elif [[ ${step} = "mos_stn_prep" ]]; then
+    export memory_gempak_gfs="GB"
```
Comment (aerorahul) [inline parm/config/gfs/config.resources]:
```suggestion
    export memory_gempak_gfs="2GB"
```

Comment (WalterKolczynski-NOAA) [reply parm/config/gfs/config.resources]: Fixed, and I went through the remaining memory lines.

**Review by aerorahul — APPROVED** [review]
looks good.

**Comment by DavidHuber-NOAA** [comment]
The Radiance Monitor failed on a `netCDF`/`HDF5` read error of `/scratch1/NCEPDEV/stmp2/Terry.McGuinness/RUNDIRS/C96C48_hybatmDA_a43d93c9/verfrad.99724/iasi_metop-c_anl`.  Examining that file, it looks like it was truncated to just the HDF5 header.  This was probably a problem with the `gdasanaldiag` job failing silently.  The log file for that job has no problems listed, but perhaps it is an issue inside one of the `MPMD` runs.  If it's possible to run this job with `KEEP_DATA=YES`, we could look at the individual log files to find the culprit.

**Comment by WalterKolczynski-NOAA** [comment]
@WenMeng-NOAA @NeilBarton-NOAA @bingfu-NOAA 

Last chance to comment before we merge.

**Pull request merged**